streams/src/main/java/org/apache/kafka/streams/TopologyDescription.java

```diff
@@ -138,6 +139,12 @@
          * @return a topic name
          */
         String topic();
+        /**
+         * The TopicNameExtractor class that this sink node using to dynamically extract the topic name to write to.
+         * Could be null if the topic name is not dynamically determined.
+         * @return the TopicNameExtractor class used get the topic name
+         */
+        Class<? extends TopicNameExtractor> topicNameExtractor();
```

Comment:
This is my first time looking at this class, but I noticed that pretty much all of the "payload" of these description nodes are strings. Should we consider returning a string here instead?

In fact, if we did that, we could consider calling `toString` on the extractor instead of returning the class name. This would allow authors of the extractors to provide more information about the extractor than just its name. This might be especially useful in the case of anonymous implementations.

Reply:
I've just realized this is more a comment on the KIP than the PR. I'll copy it there.

Reply:
Personally I think the default form of `someObjectClassname@hashcodenumber` is fine as to identify the extractor, other than a single topic name, is used, which is probably the most important illustration from `SinkNode` anyways.